### PR TITLE
lsp: Support Goto Declaration

### PR DESCRIPTION
**Pull request by coszio**
Adds support for [Goto Declaration](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_declaration) LSP command.

I am particularly interested in [this for Rust projects](https://rust-analyzer.github.io/manual.html#go-to-declaration), to be able to navigate to the place where a trait method is declared, coming from a trait method implementation. 

I noticed this was something I could do in VSCode before, but was somehow missing is Zed. Thanks to the already existing infrastructure for Goto Definition, I just followed and copy-paste-adapted it for Goto Declaration.

As a bonus, I added `ctrl-F12` and `alt-ctrl-F12` as default macOS keybindings for `GoToDeclaration` and `GoToDeclarationSplit`, respectively. They are not keybindings from another editor, but I figured they made sense to be grouped along with the other *F12 commands.

### Release Notes:

- Added "Go to declaration" editor action.
- vim: Breaking change to keybindings after introduction of the `Go to declaration` editor action. The new keybindings are the following (and can be found [here](https://zed.dev/docs/vim), alongside the other key bindings):
  - `g d` - Go to definition
  - `g D` - Go to declaration
  - `g y` - Go to type definition
  - `g I` - Go to implementation

https://github.com/user-attachments/assets/ee5c10a8-94f0-4e50-afbb-6f71db540c1b

### CODE REVIEW

**Comment by coszio** [comment]
@cla-bot check

File: docs/src/key-bindings.md [hunk 1]
```diff
@@ -218,6 +218,8 @@ See the [tasks documentation](/docs/tasks#custom-keybindings-for-tasks) for more
 | Format                           | Editor     | `⌘ + Shift + I`                 |
 | Go to definition                 | Editor     | `F12`                           |
 | Go to definition split           | Editor     | `Alt + F12`                     |
+| Go to declaration                | Editor     | `Ctrl + F12`                 |
```
Comment (diogonr) [inline docs/src/key-bindings.md:221]: nit: incorrect spacing

**Comment by mrnugget** [comment]
That's amazing! Thank you so much! I've been missing this a lot.

I hopped on the branch to fix the formatting issues and added default keybindings for Vim mode. I don't know which ones would be good for VS Code and other keymaps, but for Vim I think this one makes sense. It's also part of the standard commands defined [here](https://neovim.io/doc/user/pattern.html), like `gd` and`gD`. And I've seen it in Vim configs before. (Side-note: we should probably change the `gD` binding for `GoToImplementation` to `gi` and then use `gD` for this one here)

**Comment by mrnugget** [comment]
Changed the keybindings and essentially made a breaking-change, but I think it's worth it. The new keybindings are inspired by LazyVim: https://sourcegraph.com/github.com/LazyVim/LazyVim/-/blob/lua/lazyvim/plugins/lsp/keymaps.lua?L19-21 And other keybindings: https://sourcegraph.com/search?q=context:global+lang:lua+vim.lsp.buf.declaration&patternType=keyword&sm=0

**Pull request merged**